File: cli-ios/RemoteCoding/Models/TaskModels.swift
import SwiftUI

/// 任务状态(对应设计图的徽章与状态色)。
enum TaskStatus {
    case waitingApproval   // 等待审批
    case waitingInput      // 等待输入
    case running           // 运行中
    case completed         // 已完成
    case failed            // 执行失败

    var label: String {
        switch self {
        case .waitingApproval: return "等待审批"
        case .waitingInput:    return "等待输入"
        case .running:         return "运行中"
        case .completed:       return "已完成"
        case .failed:          return "执行失败"
        }
    }
    var color: Color {
        switch self {
        case .waitingApproval: return Theme.coral
        case .waitingInput:    return Theme.gold
        case .running:         return Theme.blue
        case .completed:       return Theme.green
        case .failed:          return Theme.coral
        }
    }
    var icon: String {
        switch self {
        case .waitingApproval: return "hourglass"
        case .waitingInput:    return "bubble.left.and.text.bubble.right"
        case .running:         return "play.circle"
        case .completed:       return "checkmark.circle"
        case .failed:          return "exclamationmark.triangle"
        }
    }
}

/// 操作按钮风格。
enum ActionStyle { case coral, gold, blue
    var bg: Color {
        switch self {
        case .coral: return Theme.coral
        case .gold:  return Theme.gold
        case .blue:  return Theme.blueBtn
        }
    }
    var fg: Color { self == .gold ? .black.opacity(0.85) : .white }
}

struct TaskItem: Identifiable, Hashable {
    let id: String
    let name: String
    let agent: String          // "Claude Code" / "Codex"
    let letter: String
    let avatarColor: Color
    let status: TaskStatus
    let detail: String         // "等待审批: npm install"
    let actionLabel: String    // "立即处理" / "回复" / "查看进度"
    let actionStyle: ActionStyle

    static func == (l: TaskItem, r: TaskItem) -> Bool { l.id == r.id }
    func hash(into h: inout Hasher) { h.combine(id) }
}

/// 通知。
enum NotiKind { case completed, waitingInput, failed
    var icon: String {
        switch self {
        case .completed:    return "checkmark.circle.fill"
        case .waitingInput: return "bubble.left.fill"
        case .failed:       return "exclamationmark.triangle.fill"
        }
    }
    var color: Color {
        switch self {
        case .completed:    return Theme.green
        case .waitingInput: return Theme.gold
        case .failed:       return Theme.coral
        }
    }
}

struct AppNotification: Identifiable {
    let id = UUID()
    let kind: NotiKind
    let title: String
    let subtitle: String
    let time: String
    let unread: Bool
}

File: cli-anr/app/src/main/java/com/aicodingremote/app/models/TaskModels.kt
package com.aicodingremote.app.models

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Chat
import androidx.compose.material.icons.filled.QuestionAnswer
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Error
import androidx.compose.material.icons.filled.HourglassEmpty
import androidx.compose.material.icons.filled.PlayCircle
import androidx.compose.material.icons.filled.Warning
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.aicodingremote.app.designsystem.Theme

/**
 * 任务状态(对应设计图的徽章与状态色)。
 * 与 iOS `TaskStatus` 一一对应。
 */
enum class TaskStatus(
    val label: String,
    val color: Color,
    val icon: ImageVector,
) {
    WaitingApproval("等待审批", Theme.coral, Icons.Default.HourglassEmpty),
    WaitingInput("等待输入", Theme.gold, Icons.Default.QuestionAnswer),
    Running("运行中", Theme.blue, Icons.Default.PlayCircle),
    Completed("已完成", Theme.green, Icons.Default.CheckCircle),
    Failed("执行失败", Theme.coral, Icons.Default.Warning),
}

/** 操作按钮风格(对位 iOS `ActionStyle`)。 */
enum class ActionStyle(val bg: Color, val fg: Color) {
    Coral(Theme.coral, Color.White),
    Gold(Theme.gold, Color(0xD9000000)), // 0.85 alpha black
    Blue(Theme.blueBtn, Color.White),
}

/** 通知大类(对位 iOS `NotiKind`)。 */
enum class NotiKind(val icon: ImageVector, val color: Color) {
    Completed(Icons.Default.CheckCircle, Theme.green),
    WaitingInput(Icons.AutoMirrored.Filled.Chat, Theme.gold),
    Failed(Icons.Default.Error, Theme.coral),
}

data class AppNotification(
    val id: String,
    val kind: NotiKind,
    val title: String,
    val subtitle: String,
    val time: String,
    val unread: Boolean,
)
